lib/apiservers/engine/backends/volume.go

```diff
@@ -117,6 +123,10 @@ func (v *Volume) VolumeCreate(name, driverName string, opts, labels map[string]s
 	if err != nil {
 		switch err := err.(type) {
 
+		case *storage.CreateVolumeConflict:
+			return result, derr.NewBadRequestError(fmt.Errorf("A volume named %s already exists. Choose a different volume name.", model.Name))
```

Comment:
According to the [docker volume create docs](https://docs.docker.com/engine/reference/api/docker_remote_api_v1.23/#/create-a-volume), only `201` and `500` are valid return status codes.